lib/ZnapZend/ZFS.pm

```diff
@@ -553,7 +557,75 @@ sub sendRecvSnapshots {
     return 1;
 }
 
+sub filterPropertyNames {
+    # This routine is a helper for getXproperties to allow easy passing of
+    # specific properties we are interested to `zfs get`, to optimize some
+    # code around that. As an input it can get either:
+    # * string that would be passed through as is (after some sanitization);
+    # * array of strings that would be concatenated into a comma-separated
+    #   string with our name-space propertyPrefix (e.g. "org.znapzend:...")
+    #   prepended if needed;
+    # * undef for defaulting to 'all'
+    # Returns a string safe to pass into `zfs get`
+    my $self = shift;
+    my $propnames = shift;
+
+    my $propertyPrefix = $self->propertyPrefix;
+
+    if (defined($propnames) && $propnames) {
+        if (ref($propnames) eq 'ARRAY') {
+            if (scalar(@$propnames) > 0) {
+                my $propstring = '';
+                foreach my $propname (@$propnames) {
+                    next if !defined($propname);
```

Comment:
```suggestion
                    next unless $propname;
```
since $propmane is probably never 0 or an empty string this should be fine

Reply:
Would a bad caller with numeric `0` be a problem?
We have an explicit check for `eq ''` below, would that suffice if we do not check `defined` here?

Reply:
$propname with "", "0", 0 is false .. question is if that is a problem

Reply:
Well, this is supposed to be a sanity-checker of sorts for whatever other authors would put into their arrays eventually, maybe generated ones. Array elements that are `undef` or `""` we skip(ed) explicitly. If evolution provides a zfs property named `0` or `org.znapzend:0` we would fail to get it with this change...

Reply:
ok

Reply:
So in the end, not applying this one, now - consciously :)